dxr/indexers.py

```diff
@@ -131,7 +131,8 @@ def file_to_index(self, path, contents):
         :arg path: A path to the file to index, relative to the tree's source
             folder
         :arg contents: What's in the file: unicode if we managed to guess an
-            encoding and decode it, str otherwise
+            encoding and decode it, str otherwise (only the first 4 kb unless
```

Comment:
The first-4K thing still doesn't strike me as useful and is potentially surprising. Either an indexer won't want to do anything with the first 4K, or they'll probably want the whole file. What do you think of simply returning None for binary files? That saves RAM in the case of huge ones, and interested plugins can still go get the file themselves based on the path and tree location and read it in segments. This also cleans up file_contents(), which no longer would need a force_read kwarg: it would just return None on binary.